hw/formal/fpv_vcf.tcl

```diff
@@ -0,0 +1,164 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script has commands to run the design using VC Formal
+
+#-------------------------------------------------------------------------
+# Analysing and elaborating the design
+#-------------------------------------------------------------------------
+# only one scr file exists in this folder
+
+analyze -format sverilog      \
+  -vcs "+define+FPV_ON -assert svaext \
+        +define+INC_ASSERT  \
+        -f [glob *.scr]"
+
+elaborate -sva $env(FPV_TOP) 
+
+#-------------------------------------------------------------------------
+# Specify clock(s) and reset(s)
+#-------------------------------------------------------------------------
+# specify the primary clocks and resets in the design.
+# use "-period" option with "create_clock" to specify the period
+# use "-sense low/high" option with "create_reset" to specify active low or high reset respectively
+#
+# note: the TL-UL protocol checker tlul_assert.sv operates on the negedge clock
+
+if {$env(FPV_TOP) == "rv_dm"} {
+  create_clock clk_i -period 100 
+  create_clock tck_i -period 100
+  create_reset rst_ni -sense low
+  create_reset trst_ni -sense low
+} elseif {$env(FPV_TOP) == "spi_device"} {
+  create_clock clk_i -period 100
+  create_clock cio_sck_i -period 100
+  create_reset rst_ni -sense low
+  create_reset cio_csb_i -sense high 
+} elseif {$env(FPV_TOP) == "usb_fs_nb_pe"} {
+  create_clock clk_48mhz_i -period 100
```

Comment:
As the name indicates, this is a 48MHz clock; period should be set accordingly?

Reply:
Hi Sri,

The exact frequency doesn’t matter for Formal verification. The usual convention is to first run all the clocks with same frequency to fix the design holes and keep the period simple numbers (to help with debug).
If you had 2 or more clocks then you would use this to set relative frequency, the next step is to maintain clock ratio right with whole numbers and then run (which not a lot of users do).

Let's know if you have more questions.